.github/workflows/nightly_build.yml

```diff
@@ -26,10 +26,20 @@ jobs:
           aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
           aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
           aws-region: us-east-2
+      - name: configure gcloud credentials
+        uses: google-github-actions/setup-gcloud@master
+        with:
+          project_id: ${{ secrets.GCP_PROJECT_ID }}
+          service_account_key: ${{ secrets.GCP_SA_KEY }}
+          export_default_credentials: true
       - name: download data
         run: dvc pull data/cats_dogs.dvc
       - name: download existing results
         run: aws s3 cp --recursive s3://dvc-bench/latest_results results
+      - name: configure dvc-bench
+        env:
+          CONFIG_JSON: ${{ secrets.CONFIG_JSON }}
+        run: echo "$CONFIG_JSON" > config.json
```

Comment:
What is a purpose of this JSON? I see it is already created. Why do we want to override it?

Reply:
The default one only contain s3, the one we have in the CI contains others depending on the authentication methods we'll add (currently only s3 and gs).

Reply:
So, lets assume we support multiple remotes - every time we will be adding new entry to the config (or rotate the credentials), we will need to recreate the JSON, add new entry and override existing secret, right? I wonder if it wont be too much work (also we would need to keep this JSON somewhere). What do we need to keep secret? Authentication credentials and thats probably all, right? Maybe we should have a config for non-critical values and keep critical ones in secrets? Access to particular secrets could be based on `remote_type` param inside benchmarks?

Reply:
> So, lets assume we support multiple remotes - every time we will be adding new entry to the config (or rotate the credentials), we will need to recreate the JSON, add new entry and override existing secret, right?

That depends;
  - For rotating, if the credentials can exposed (e.g like `~/.aws/credentials`) we don't need to change anything on the config, since it will contain just the URL for those providers.
  - For adding a new entry, I assume this is going to be an extremely rare situation, and if so we'd just add a new entry to the json to denote the URL

> What do we need to keep secret?

We could keep the URLs within repo, though the JSON sometimes might contain other secrets (any sort of parameter that `dvc remote modify` can receive) so I think it is better to keep it as a secret.

> Access to particular secrets could be based on remote_type param inside benchmarks?

We could simply concanate 2 JSON files in the initial part though I don't know whether this is something we need since amending something to a JSON file isn't that much of a trouble. Wdyt?

Reply:
I think we  should not create config json as a secret.

1. Even if updates/edits will be rare, we will need to keep the config copy somewhere, just to see how it did look before update. Since it can contain actual secrets, we cannot keep it in this repo. 
2. In the unlikely case of secret deletion and loosing our "outside" copy, one will need to read through the code to decipher what was the `config` structure.

I think that in secrets we should just keep the actual secrets and access them in code via env variables - that way we can say which benchmark requires which secret and the benchmarks for different clouds won't rely on same secret.

Reply:
I like it! Seems to me it joins the pros of both approaches, we keep config in one place, the structure is public and we can easily see what secrets are actually required.